src/odemis/driver/hamamatsurx.py

```diff
@@ -794,6 +821,20 @@ def __init__(self, name, role, parent, daemon=None, **kwargs):
         # TODO do we need: Burst Mode, Setting, Trig. Mode, delay B ??? as read only... plus MD!?
         # self.delayB = model.VigilantAttribute(self.parent.DevParamGet(self.location, "Delay B"), readonly=True)
 
+    # override HwComponent.updateMetadata
+    def updateMetadata(self, md):
+
+        if model.MD_TIME_RANGE_TO_DELAY in md:
+            for timeRange, delay in md[model.MD_TIME_RANGE_TO_DELAY].iteritems():
```

Comment:
Just use `.items()`. It slightly less efficient (on long dicts), but it has the advantage of directly working in Python 3.